registry/storage/driver/testsuites/testsuites.go

```diff
@@ -527,30 +527,54 @@ func (suite *DriverSuite) TestList(c *check.C) {
 	// 3. Ensure that we only respond to directory listings that end with a slash (maybe?).
 }
 
-// TestMove checks that a moved object no longer exists at the source path and
+// TestMoveSmall checks that a moved object no longer exists at the source path and
 // does exist at the destination.
-func (suite *DriverSuite) TestMove(c *check.C) {
-	contents := randomContents(32)
-	sourcePath := randomPath(32)
-	destPath := randomPath(32)
+func (suite *DriverSuite) TestMoveSmall(c *check.C) {
+	suite.testMoveFile(c, 32)
+}
 
-	defer suite.deletePath(c, firstPart(sourcePath))
-	defer suite.deletePath(c, firstPart(destPath))
+// TestMoveLarge tests that a large file is moved to the storage
+// driver safely.
+func (suite *DriverSuite) TestMoveLarge(c *check.C) {
+	if testing.Short() {
+		c.Skip("Skipping test in short mode")
+	}
+
+	suite.testMoveFile(c, 5*1024*1024*1024)
+}
+
+func (suite *DriverSuite) testMoveFile(c *check.C, fileSize int64) {
+	srcFile := randomPath(32)
+	destFile := randomPath(32)
 
-	err := suite.StorageDriver.PutContent(suite.ctx, sourcePath, contents)
+	defer suite.StorageDriver.Delete(suite.ctx, firstPart(srcFile))
+	defer suite.StorageDriver.Delete(suite.ctx, firstPart(destFile))
+
+	srcDigester := digest.Canonical.New()
+
+	contents := newRandReader(fileSize)
+	tr := io.TeeReader(contents, srcDigester.Hash())
+	written, err := suite.StorageDriver.WriteStream(suite.ctx, srcFile, 0, tr)
 	c.Assert(err, check.IsNil)
+	c.Assert(written, check.Equals, fileSize)
 
-	err = suite.StorageDriver.Move(suite.ctx, sourcePath, destPath)
+	err = suite.StorageDriver.Move(suite.ctx, srcFile, destFile)
 	c.Assert(err, check.IsNil)
 
-	received, err := suite.StorageDriver.GetContent(suite.ctx, destPath)
+	reader, err := suite.StorageDriver.ReadStream(suite.ctx, destFile, 0)
 	c.Assert(err, check.IsNil)
-	c.Assert(received, check.DeepEquals, contents)
+	defer reader.Close()
 
-	_, err = suite.StorageDriver.GetContent(suite.ctx, sourcePath)
+	destDigester := digest.Canonical.New()
+	written, err = io.Copy(destDigester.Hash(), reader)
+	c.Assert(written, check.Equals, fileSize)
+	c.Assert(srcDigester.Digest(), check.DeepEquals, destDigester.Digest())
```

Comment:
`srcDigester.Digest() == dstDigester.Digest()`. No need to rely on deep equals here.